### PR TITLE
1.13 / 1.14 update

### PR DESCRIPTION
**Pull request by jascotty2**
Converts to 1.13 block data methods, adds compatibility with 1.14 (Still compatible with older versions)

### CODE REVIEW

**Comment by Pyario** [comment]
Can you send me the .jar file for 1.14 update? :D

**Comment by jascotty2** [comment]
@Pyario https://github.com/jascotty2/AreaShop/raw/1.14/AreaShop_2.6.0_for_1.14.jar

**Comment by Pyario** [comment]
thanks.

**Comment by the-og-gear** [comment]
This PR has bloated the commit with the change of `i++` in for loops to `++i`

Is this some new standard I'm missing out on, or is this personal preference when reading/writing? If it is the latter, please do not do this in PRs, if the for loops are already established. As far as I know, this offers no performance difference and the compiler interprets them both the same in such simple expressions.

File: AreaShop/pom.xml [hunk 1]
```diff
@@ -150,6 +150,14 @@
             <artifactId>bstats-bukkit</artifactId>
             <version>1.1</version>
         </dependency>
+
+		<!-- https://mvnrepository.com/artifact/com.googlecode.json-simple/json-simple -->
+		<dependency>
+			<groupId>com.googlecode.json-simple</groupId>
+			<artifactId>json-simple</artifactId>
+			<version>1.1.1</version>
+		</dependency>
```
Comment (the-og-gear) [inline AreaShop/pom.xml]: This dependency is unused, bloats the file.

File: AreaShop/src/main/java/me/wiefferink/areashop/regions/GeneralRegion.java [hunk 2]
```diff
@@ -1454,22 +1453,18 @@ public void runCommands(CommandSender sender, List<String> commands) {
...
 			}
 			boolean printed = false;
 			if(!result) {
 				printed = true;
-				if(error != null) {
-					AreaShop.warn("Command execution failed, command=" + command + ", error=" + error + ", stacktrace:");
-					AreaShop.warn(stacktrace);
-					AreaShop.warn("--- End of stacktrace ---");
+				if(e != null) {
+					AreaShop.warn(e, "Command execution failed, command=" + command);
```
Comment (the-og-gear) [inline AreaShop/src/main/java/me/wiefferink/areashop/regions/GeneralRegion.java]: This is all essentially the exact same, why change it?

File: AreaShop/src/main/java/me/wiefferink/areashop/features/signs/RegionSign.java [hunk 3]
```diff
@@ -139,7 +140,7 @@ public boolean update() {
 		// Get the lines
 		String[] signLines = new String[4];
 		boolean signEmpty = true;
-		for(int i = 0; i < 4; i++) {
+		for(int i = 0; i < 4; ++i) {
```
Comment (the-og-gear) [inline AreaShop/src/main/java/me/wiefferink/areashop/features/signs/RegionSign.java]: Very unnecessary change.

File: AreaShop/src/main/java/me/wiefferink/areashop/features/signs/RegionSign.java [hunk 4]
```diff
@@ -207,7 +229,7 @@ public boolean needsPeriodicUpdate() {
 		}
 		ConfigurationSection stateConfig = signConfig.getConfigurationSection(getRegion().getState().getValue().toLowerCase());
 		// Check the lines for the timeleft tag
-		for(int i = 1; i <= 4; i++) {
+		for(int i = 1; i <= 4; ++i) {
```
Comment (the-og-gear) [inline AreaShop/src/main/java/me/wiefferink/areashop/features/signs/RegionSign.java]: Very unnecessary change.

File: AreaShop/src/main/java/me/wiefferink/areashop/tools/GithubUpdateCheck.java [hunk 5]
```diff
@@ -1,19 +1,20 @@
 package me.wiefferink.areashop.tools;
 
-import org.apache.commons.lang.StringUtils;
-import org.apache.commons.lang.exception.ExceptionUtils;
-import org.bukkit.plugin.Plugin;
-import org.bukkit.scheduler.BukkitRunnable;
+//import com.google.gson.Gson;
+//import com.google.gson.GsonBuilder;
```
Comment (the-og-gear) [inline AreaShop/src/main/java/me/wiefferink/areashop/tools/GithubUpdateCheck.java]: Unused, so commented? Why leave them in?

File: AreaShop/src/main/java/me/wiefferink/areashop/tools/GithubUpdateCheck.java [hunk 6]
```diff
@@ -110,7 +111,7 @@ public void run() {
 						try(BufferedReader reader = new BufferedReader(new InputStreamReader(conn.getInputStream()))) {
 							String response = reader.readLine();
 							debug("Response:", response);
-
+							//gson.fromJson(response, );
```
Comment (the-og-gear) [inline AreaShop/src/main/java/me/wiefferink/areashop/tools/GithubUpdateCheck.java]: Offers no purpose with the blank argument. Has no need to be in the file.

File: AreaShop/pom.xml (hunk 1 above)
Comment (jascotty2) [reply AreaShop/pom.xml]: This is required as of 1.14 api changes in order to use org.json.simple.JSONObject
Please note: https://www.spigotmc.org/threads/bukkit-craftbukkit-spigot-bungeecord-1-14-2-development-builds.369724/

Also, please actually check if a dependency is unused before reporting that it is?

File: AreaShop/src/main/java/me/wiefferink/areashop/regions/GeneralRegion.java (hunk 2 above)
Comment (jascotty2) [reply AreaShop/src/main/java/me/wiefferink/areashop/regions/GeneralRegion.java]: I'm reordering it here to avoid importing an entire library just to use one function (ExceptionUtils)

File: AreaShop/src/main/java/me/wiefferink/areashop/features/signs/RegionSign.java (hunk 3 above)
Comment (jascotty2) [reply AreaShop/src/main/java/me/wiefferink/areashop/features/signs/RegionSign.java]: Did a little digging, and apparently JRE 8 actually does optimize for loops written like that. Neat. 
Personally, I tend to lean back to my older c++ coding habits where post increment is a waste of resources if the returned value is not used xD

**Comment by jascotty2** [comment]
@Pyario Fix for `/as linksigns` - really only needed for mass-adding, but just in case ;)
https://github.com/jascotty2/AreaShop/blob/1.14/AreaShop_2.6.0_for_1.14-2.jar

**Comment by james090500** [comment]
Could you possibly update this PR with a fix to  /as addsign stall1

https://pastebin.com/bxnQNiyH

**Comment by jascotty2** [comment]
@james090500 Yep! 
https://github.com/jascotty2/AreaShop/blob/1.14/AreaShop_2.6.0_for_1.14-3.jar

Thanks for the heads-up! :)

**Comment by james090500** [comment]
@jascotty2 

Thanks very much! I'll keep you updated of any bugs.

**Comment by jascotty2** [comment]
Implemented via adapters instead of in-place in `4cda8a9` - closing in lieu of official version
